tools/cpp/CROSSTOOL.tpl

```diff
@@ -990,12 +978,140 @@ toolchain {
       action: 'c-compile'
       action: 'c++-compile'
       flag_group {
-        flag: "/O2"
+        flag: "/O2" # Implies /Og /Oi /Ot /Oy /Ob2 /Gs /GF /Gy
         flag: "/DNDEBUG"
       }
     }
   }
 
+  feature {
+    name: 'treat_warnings_as_errors'
+    flag_set {
+      action: 'c-compile'
+      action: 'c++-compile'
+      flag_group {
+        flag: "/WX"
+      }
+    }
+  }
+
+  # Enable C++ exception by default.
+  feature {
+    name: 'default_exceptions'
+    enabled: true
+    flag_set {
+      action: 'c++-compile'
+      with_feature: {
+        not_feature: 'no_exceptions'
+        not_feature: 'use_exceptions'
+      }
+      flag_group {
+        flag: "/D_HAS_EXCEPTIONS=1"
+        flag: "/EHsc"
+      }
+    }
+  }
+
+  feature {
+    name: 'use_exceptions'
```

Comment:
Could you elaborate on why we need both default_exceptions and use_exceptions? Feel free to add multiline comments describing intended usage

Reply:
Done. You can think of `default_exceptions` as `c++default` in https://github.com/tensorflow/tensorflow/blob/master/third_party/toolchains/clang6/CROSSTOOL.tpl#L276